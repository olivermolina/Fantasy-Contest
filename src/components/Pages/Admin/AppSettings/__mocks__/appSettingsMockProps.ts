export const appSettingsMockProps = {
  appSettings: {
    BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: '25',
    STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
    NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
    MAX_MATCH_DEPOSIT_AMOUNT: '100',
    DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
    RELOAD_BONUS_TYPE: 'FLAT',
    RELOAD_BONUS_AMOUNT: '0',
    MAX_RETENTION_BONUS: '50',
    RETENTION_BONUS_WEEKLY_CHANCE: '1',
    RETENTION_BONUS_MATCH_MULTIPLIER: '1',
    REFERRAL_CREDIT_AMOUNT: '25',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    MIN_BET_AMOUNT: '1',
    MAX_BET_AMOUNT: '50',
    MAX_MARKET_ODDS: '150',
    MIN_MARKET_ODDS: '-150',
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REPEAT_ENTRIES_LIMIT: '1',
  },
  onSubmit: () => console.log('onSubmit'),
};
