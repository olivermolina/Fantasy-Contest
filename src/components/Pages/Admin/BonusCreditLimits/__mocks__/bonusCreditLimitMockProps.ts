import { BetStakeType } from '@prisma/client';

export const bonusCreditLimitMockProps = {
  onSubmit: () => console.log('submit'),
  data: {
    signupFreeEntry: false,
    bonusCreditFreeEntryEquivalent: 50,
    bonusCreditLimits: [
      {
        contestCategoryId: '1',
        numberOfPicks: 2,
        id: '30387acd-660c-46f0-a9e6-af87c007e6fb',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 25,
      },
      {
        contestCategoryId: '2',
        numberOfPicks: 3,
        id: '316555e5-b447-468a-b57f-381bbf64919a',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 50,
      },
      {
        contestCategoryId: '3',
        numberOfPicks: 4,
        id: 'fc063fcd-1a25-46e2-9361-d947eec1f9a9',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 40,
      },
      {
        contestCategoryId: '4',
        numberOfPicks: 5,
        id: 'a2542740-e90f-4763-8a3a-f7afba68898c',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 20,
      },
    ],
  },
  openAddUsersFreeEntryConfirmDialog: () => console.log('open'),
};
