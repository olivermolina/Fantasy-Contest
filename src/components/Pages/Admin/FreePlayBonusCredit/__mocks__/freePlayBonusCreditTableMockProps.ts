import { PaymentStatusCode } from '~/constants/PaymentStatusCode';
import { TransactionType } from '@prisma/client';

export const freePlayBonusCreditTableMockProps = {
  onSubmitDeletePlayFreeCredit: () => console.log('delete'),
  data: [
    {
      id: '1',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.COMPLETE,
      type: TransactionType.CREDIT,
    },
    {
      id: '2',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.CANCELLED,
      type: TransactionType.DEBIT,
    },
  ],
  totalBonusCreditBalance: 0,
};
