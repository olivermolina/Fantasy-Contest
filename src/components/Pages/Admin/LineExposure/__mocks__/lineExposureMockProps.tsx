import { League } from '@prisma/client';

export const lineExposureMockProps = {
  data: [
    {
      id: '1',
      name: 'LeBron James',
      category: 'Points',
      league: League.NBA,
      status: 'Scheduled',
      matchup: 'LAL @ DEN',
      total: 35,
      gamedate: '2023-01-23',
      gametime: '03:00:00 PM',
      underOdds: 100,
      overOdds: -120,
      overExposure: 4,
      underExposure: 2,
      betLegs: [
        {
          id: 'ticket-id',
          betId: 'bet-id',
          entryDate: '2023-01-22',
          payout: 100,
          stake: 10,
          type: 'OVER_ODDS',
          user: {
            id: 'user-1',
            firstName: 'John',
            lastName: 'Doe',
            username: 'jdoe',
          },
          status: 'WIN',
        },
      ],
    },
  ],
  setDateRange: () => console.log('setDateRange'),
  dateRange: null,
  selectedLeague: League.NBA,
  setSelectedLeague: () => console.log('setSelectedLeague'),
};
