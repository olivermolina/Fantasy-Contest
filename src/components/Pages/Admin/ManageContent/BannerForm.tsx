import React, { useEffect } from 'react';
import {
  Box,
  FormHelperText,
  IconButton,
  MenuItem,
  Select,
  Tab,
  Tabs,
  TextField,
} from '@mui/material';
import Button from '@mui/material/Button';
import { Controller, useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import * as z from 'zod';
import { ManageBannerRowModel } from './ManageBanners';
import { NEW_USER_ID } from '~/constants/NewUserId';
import CloseIcon from '@mui/icons-material/Close';
import { BannerFormValidationSchema } from '~/schemas/BannerFormValidationSchema';
import FormControl from '@mui/material/FormControl';
import { AppSettings } from '@prisma/client';

interface TabPanelProps {
  children?: React.ReactNode;
  index: number;
  value: number;
}

function TabPanel(props: TabPanelProps) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && <>{children}</>}
    </div>
  );
}

function a11yProps(index: number) {
  return {
    id: `banner-tab-${index}`,
    'aria-controls': `banner-tabpanel-${index}`,
  };
}

/**
 * Defines the form fields for user form.
 */
export type BannerFormInputs = z.infer<typeof BannerFormValidationSchema>;

/**
 * Represents the properties of `UserForm` component.
 */
interface Props {
  banner: ManageBannerRowModel;
  onSubmit: (data: BannerFormInputs) => void;
  closeForm: () => void;
  appSettings: AppSettings[];
}

/**
 * Defines the `UserForm` component.
 */
export default function BannerForm(props: Props) {
  const { banner, closeForm } = props;
  const {
    register,
    handleSubmit,
    formState: { errors },
    control,
    reset,
  } = useForm<BannerFormInputs>({
    resolver: zodResolver(BannerFormValidationSchema),
  });

  const [tabValue, setTabValue] = React.useState(0);

  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setTabValue(newValue);
  };

  /**
   * Resets the form fields when `banner` prop changes.
   */
  useEffect(() => {
    reset({
      id: banner.id,
      text: banner.text,
      priority: banner.priority,
      appSettingId: banner.appSettingId,
    });
  }, [banner]);

  // Reset the tab value when there are errors
  useEffect(() => {
    if (Object.keys(errors).length > 0) {
      // If there are errors in the first tab, set the tab value to 0
      if (errors.text || errors.priority) {
        setTabValue(0);
      } else {
        setTabValue(1);
      }
    }
  }, [errors]);

  return (
    <form onSubmit={handleSubmit(props.onSubmit)}>
      <div
        className={
          'flex justify-between bg-selected p-4 text-xl text-white items-center'
        }
      >
        <p className={'font-bold'}>
          {banner.id === NEW_USER_ID ? 'Add ' : 'Edit '} User
        </p>
        <IconButton sx={{ color: 'inherit', mt: -1 }} onClick={closeForm}>
          <CloseIcon />
        </IconButton>
      </div>
      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
        <Tabs
          value={tabValue}
          onChange={handleChange}
          aria-label="User Tabs"
          variant="fullWidth"
        >
          <Tab label="Account" {...a11yProps(0)} />
        </Tabs>
      </Box>
      <TabPanel value={tabValue} index={0}>
        <div className={'flex flex-col  divide-y divide-dashed justify-center'}>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>ID</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('id')}
                disabled
              />
            </div>
          </div>

          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Text</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('text')}
                error={!!errors?.text}
                helperText={errors?.text?.message}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Priority</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                type="number"
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('priority')}
                error={!!errors?.priority}
                helperText={errors?.priority?.message}
              />
            </div>
          </div>

          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>AppSetting</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <Controller
                name="appSettingId"
                control={control}
                defaultValue={''}
                render={({ field }) => (
                  <FormControl
                    fullWidth
                    error={!!errors?.appSettingId}
                    size={'small'}
                  >
                    <Select size={'small'} fullWidth {...field}>
                      {props.appSettings.map((appSetting) => (
                        <MenuItem key={appSetting.id} value={appSetting.id}>
                          {appSetting.name} &mdash;
                          <span className={'italic text-gray-500 pl-1'}>
                            ${appSetting.value}
                          </span>
                        </MenuItem>
                      ))}
                    </Select>

                    {errors?.appSettingId?.message ? (
                      <FormHelperText>
                        {errors?.appSettingId?.message}
                      </FormHelperText>
                    ) : null}
                  </FormControl>
                )}
              />
            </div>
          </div>
        </div>
      </TabPanel>

      <div className={'flex flex-row justify-between p-4'}>
        <Button variant={'outlined'} onClick={closeForm}>
          Cancel
        </Button>
        <Button variant={'outlined'} type={'submit'}>
          Save
        </Button>
      </div>
    </form>
  );
}
