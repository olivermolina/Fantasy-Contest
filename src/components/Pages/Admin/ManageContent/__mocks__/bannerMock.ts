import { AppSettingName } from '@prisma/client';

export const bannerMock = {
  id: '1',
  text: 'test',
  priority: 1,
  appSettingId: '1',
  appSetting: {
    id: '1',
    name: AppSettingName.BONUS_CREDIT_FREE_ENTRY_EQUIVALENT,
    value: '1',
  },
};
