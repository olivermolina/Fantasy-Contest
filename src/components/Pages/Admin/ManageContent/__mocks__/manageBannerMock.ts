export const manageBannerMock = {
  banners: [],
  openBannersForm: jest.fn(),
  contentSettings: {
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    HOMEPAGE_HEADING_1: 'Heading 1',
    HOMEPAGE_HEADING_2: 'Heading 2',
  },
  onSubmit: () => console.log('onSubmit'),
};
