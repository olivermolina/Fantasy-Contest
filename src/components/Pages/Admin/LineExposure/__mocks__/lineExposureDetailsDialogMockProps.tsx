import { lineExposureMockProps } from './lineExposureMockProps';

export const lineExposureDetailsDialogMockProps = {
  row: lineExposureMockProps.data[0],
  open: false,
  handleClose: () => console.log(close),
};
