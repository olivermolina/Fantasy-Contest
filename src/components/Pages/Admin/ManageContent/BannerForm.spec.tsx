import { render } from '@testing-library/react';
import BannerForm from './BannerForm';
import { bannerMock } from './__mocks__/bannerMock';

describe('BannerForm', () => {
  it('should render correctly', () => {
    const mockProps = {
      banner: bannerMock,
      closeForm: jest.fn(),
      onSubmit: jest.fn(),
      appSettings: [],
    };

    const { container } = render(<BannerForm {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
