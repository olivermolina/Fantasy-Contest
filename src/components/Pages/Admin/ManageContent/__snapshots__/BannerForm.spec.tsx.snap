// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerForm should render correctly 1`] = `
<div>
  <form>
    <div
      class="flex justify-between bg-selected p-4 text-xl text-white items-center"
    >
      <p
        class="font-bold"
      >
        Edit 
         User
      </p>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1qxc83e-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiBox-root css-1gsv261"
    >
      <div
        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            aria-label="User Tabs"
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-controls="banner-tabpanel-0"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1ujykiq-MuiButtonBase-root-MuiTab-root"
              id="banner-tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              Account
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-0"
      id="simple-tabpanel-0"
      role="tabpanel"
    >
      <div
        class="flex flex-col  divide-y divide-dashed justify-center"
      >
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            ID
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id=":r0:"
                  name="id"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            Text
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r1:"
                  name="text"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            Priority
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r2:"
                  name="priority"
                  type="number"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            AppSetting
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="mui-component-select-appSettingId"
                  class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-component-select-appSettingId"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="appSettingId"
                  tabindex="-1"
                  value="1"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-between p-4"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Cancel
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Save
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
